### ISSUE TITLE
Segfault on close

### ISSUE THREAD

**Issue by MrAureliusR** [MayaPosch/MQTTCute#4]
This has been present in all versions I use. Upon closing, we see this in the console:
![screenshot from 2019-03-03 17 09 53](https://user-images.githubusercontent.com/41264289/53702654-449e5180-3dd7-11e9-8f44-f618558e8b4f.png)
Also, the "Failed to connect to server" popup occurs.

However, functionality is not affected in any way.

This is on Ubuntu Budgie 18.10. 

$ gcc --version
gcc (Ubuntu 8.2.0-7ubuntu1) 8.2.0
$ qmake --version
QMake version 3.1
Using Qt version 5.11.1 in /usr/lib/x86_64-linux-gnu

**Comment by MayaPosch** [comment]
Yes, the current shutdown and clean-up of the MQTT library is still rather messy. The code will be refactored to prevent this issue from occurring. Please standby :)

**Comment by MayaPosch** [comment]
A new release (v0.3-alpha) has been made available. As the crashes and stability issues originated in the libmosquitto library, it's expected that by swapping it with NymphMQTT such crashes and stability issues should be much reduced.

Feel free to try the new code and let me know if you need any assistance with the new dependencies (NymphMQTT and its dependencies).
